midterm/Favorites.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+	useFocusEffect,
+	useNavigation,
+} from "@react-navigation/native";
+import { useState } from "react";
+import {
+	SafeAreaView,
+	Text,
+	StyleSheet,
+	View,
+	Image,
+} from "react-native";
+import { TouchableOpacity } from "react-native";
+import { useAsyncStorage } from "@react-native-async-storage/async-storage";
+import React from "react";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { favoriteItem } from "./types/favoriteItem";
+import { MidtermTabParamList } from "./types/MidtermTabParamList";
+import { ScrollView } from "react-native-gesture-handler";
+
+export const Favorites = () => {
+	const [favoritesList, setFavoritesList] = useState<favoriteItem[]>([]);
+	const { getItem } = useAsyncStorage("@storage_key");
+
+	useFocusEffect(
+		React.useCallback(() => {
+			retrieveFavorites();
+		}, [])
+	);
+	const retrieveFavorites = async () => {
+		console.log("retrieveFavorites");
+		const favs = await getItem();
+
+		if (favs !== null) {
+			setFavoritesList(JSON.parse(favs));
+			//put in if statement to satisfy typescript
+		}
+	};
+
+	type FavNavProp = StackNavigationProp<MidtermTabParamList, "Favorites">;
```

Comment:
any `type` or `interface` should live outside of the component since it will never change.